src/data_provider/src/ports/portSolarisHelper.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Wazuh SysInfo
+ * Copyright (C) 2015, Wazuh Inc.
+ * March 17, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stropts.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include <memory>
+
+#include "portSolarisHelper.hpp"
+
+int PortSolarisHelper::tcpConnEntrySize;
+int PortSolarisHelper::tcp6ConnEntrySize;
+int PortSolarisHelper::udpEntrySize;
+int PortSolarisHelper::udp6EntrySize;
+
+void PortSolarisHelper::mibGetItems(int sd, std::vector<mib_item_t>& items)
+{
+    /*
+     * buf is an automatic for this function, so the
+     * compiler has complete control over its alignment;
+     * it is assumed this alignment is satisfactory for
+     * it to be casted to certain other struct pointers
+     * here, such as struct T_optmgmt_ack * .
```

Comment:
Could you rephrase this? what is "alignment"?

Reply:
I leave this original comment, to accomplish